frontend/src/components/kl-add-item.js

```diff
@@ -22,17 +21,10 @@ export default {
     },
     methods: {
         addItem: function () {
-            let item = clone(this.item);
-            this.$http.post(API.ITEMS, item).then(response => {
-                item._id = response.body.added_id;
-                this.$emit('addItem', item);
-            }).catch(err => {
-                if (err.status === 400 && err.body.code === 1) {
-                    let renderedErrors = renderValidationErrors(err.body.errors);
-                    renderedErrors.forEach(error => toastr.error(this.$t('errors.' + error.id, error.props)));
-                } else
-                    toastr.error(this.$t('errors.default'));
-            });
+            if (localStorage.getItem('token'))
+                addItem.remote.addItem(this);
+            else
+                addItem.local.addItem(this);
```

Comment:
Really really nice! Pretty elegant! :+1: 

Few comments tho:
- why not `addItem.remote(this)`? writing addItem twice seems redundant
- `localStorage.getItem('token')` should be encapsulated in separate function like `isAuthenticated`
- here your component have control on how the item is stored - locally or remotely. But should the component really care about it? May be introduce `addItem.default` alongside with `addItem.local` and `addItem.remote`, which would choose the proper method based on authentication?